detector/vuls2/vendor.go

```diff
@@ -173,94 +144,287 @@ func selectFixedIn(family string, fixed []string) string {
 	}
 }
 
-func advisoryReference(family string, a *advisoryTypes.Advisory, r referenceTypes.Reference) *models.Reference {
-	switch family {
-	case constant.RedHat, constant.CentOS, constant.Rocky:
-		if a != nil && strings.Contains(r.URL, string(a.Content.ID)) {
-			return &models.Reference{
-				Link: r.URL,
-				Source: func() string {
-					switch family {
-					case constant.RedHat, constant.CentOS:
-						return "RHSA"
-					case constant.Rocky:
-						return "RLSA"
-					default:
-						return r.Source
-					}
-				}(),
-				RefID: string(a.Content.ID),
+func comparePackStatus(a, b packStatus) (int, error) {
+	if a.status.Name != b.status.Name {
+		return 0, xerrors.Errorf("Package names are different. a: %s, b: %s", a.status.Name, b.status.Name)
+	}
+
+	if a.rangeType != vcAffectedRangeTypes.RangeTypeUnknown && b.rangeType != vcAffectedRangeTypes.RangeTypeUnknown && a.rangeType != b.rangeType {
+		return 0, xerrors.Errorf("Range types are different. a: %s, b: %s", a.rangeType, b.rangeType)
+	}
+
+	return cmp.Or(
+		func() int {
+			switch {
+			case a.status.NotFixedYet && b.status.NotFixedYet:
+				return 0
```

Comment:
```suggestion
```

Can be merged into the default clause?

Reply:
fixed: https://github.com/future-architect/vuls/compare/e3a15408d4270ce78b61ed041c8ac67874cc3c64..e9a86d8f93fcdb9afbf98faac678957929102ad9